warehouse/forklift/legacy.py

```diff
@@ -1065,6 +1069,56 @@ def file_upload(request):
                 k: h.hexdigest().lower() for k, h in metadata_file_hashes.items()
             }
 
+        # Check that if the file was uploaded with attestations, verification
+        # passes
+        if "attestations" in request.POST:
+            publisher = request.oidc_publisher
+            if not publisher or not publisher.publisher_name == "GitHub":
+                raise _exc_with_message(
+                    HTTPBadRequest,
+                    "Attestations are currently only supported when using Trusted "
+                    "Publishing with GitHub Actions.",
+                )
+            try:
+                attestations = TypeAdapter(list[Attestation]).validate_json(
+                    request.POST["attestations"]
+                )
+                verification_policy = publisher.publisher_verification_policy(
+                    request.oidc_claims
+                )
+                for attestation_model in attestations:
+                    # For now, attestations are not stored, just verified
+                    attestation_model.verify(
+                        Verifier.production(),
```

Comment:
Not an issue with this PR per se, but the presence of this parameter is making me think that the current `verify(...)` API is a little wonky: maybe rather than taking a `Verifier` it should just instantiate one directly, since in practice we expect this to always be `Verifier.production()`?

Reply:
We do use `Verifier.staging()` in the `pypi-attestation-models` unit tests. If that's not strictly needed, we can remove it. Otherwise, we could move the parameter to the end of the parameter list and give it a default value of `Verifier.production()`

Reply:
> We do use `Verifier.staging()` in the `pypi-attestation-models` unit tests. If that's not strictly needed, we can remove it. Otherwise, we could move the `verifier` parameter to the end of the parameter list and give it a default value of `Verifier.production()`

Ah yeah, hmm. This is probably more trouble than it's currently worth 🙂